packages/ui/src/services/mapping.service.ts

```diff
@@ -88,11 +110,18 @@ export class MappingService {
     }, [] as MappingItem[]);
   }
 
-  private static hasStaleSourceDocument(expression: string, documentType: DocumentType, documentId: string) {
+  private static hasStaleSourceDocument(
+    expression: string,
+    documentType: DocumentType,
+    documentId: string,
+    documentDefinitionType?: DocumentDefinitionType,
+  ) {
     const stalePath = XPathService.extractFieldPaths(expression).find((xpath) => {
       return (
         (documentType === DocumentType.SOURCE_BODY && !xpath.documentReferenceName) ||
-        (documentType === DocumentType.PARAM && xpath.documentReferenceName === documentId)
+        (documentType === DocumentType.PARAM && documentDefinitionType === DocumentDefinitionType.JSON_SCHEMA
+          ? xpath.documentReferenceName === `${documentId}-x`
```

Comment:
@igarashitm 
Is this the correct way to handle `documentReferenceName` in a JSON schema?  like with `${documentId}-x`?